azurerm/internal/services/managedservices/resource_arm_lighthouse_assignment.go

```diff
@@ -0,0 +1,194 @@
+package managedservices
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/managedservices/mgmt/2019-06-01/managedservices"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmLighthouseAssignment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmLighthouseAssignmentCreateUpdate,
+		Read:   resourceArmLighthouseAssignmentRead,
+		Update: resourceArmLighthouseAssignmentCreateUpdate,
+		Delete: resourceArmLighthouseAssignmentDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"registration_assignment_id": {
```

Comment:
I think we could just name this as `name` like `azurerm_policy_assignment` does. A resource itself would have an ID, and naming this as some ID may cause confusion. What do you think?

Reply:
@ArcturusZhang  I reviewed the api again and I agree with naming it from `registration_assignment_id` to `name`. Is there any other feedback which I can address along with this change?

Reply:
I have changed this to `name`. It can't be anything other than uuid.